torch/utils/data/datapipes/datapipe.py

```diff
@@ -93,11 +94,10 @@ def class_function(cls, enable_df_api_tracing, source_dp, *args, **kwargs):
     def __getstate__(self):
         if IterDataPipe.getstate_hook is not None:
             return IterDataPipe.getstate_hook(self)
-        # TODO: Fix `dill` circular dependency - https://github.com/pytorch/data/issues/237
```

Comment:
I don't think this can be removed. `torchvision` is going to use methods bound to `IterDataPipe`'s. Thus, if someone has `dill` installed, it still won't work, right?

Reply:
It should work if this condition `if isfunction(v) and not ismethod(v):` works as I intend, because the addition of `not ismethod(v)` will prevent methods bound to `IterDataPipe` from going through the custom serialization with `dill`, which I believe will be fine as long as the methods are pickable.